connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/operation/ParameterDescriptor.java

```diff
@@ -56,4 +56,29 @@ public boolean isRequired() {
       return required;
     }
   }
+
+  final class Header<T> implements ParameterDescriptor {
+
+    private final String name;
+    private final Class<T> type;
+    private final boolean required;
```

Comment:
nit: this can be a record I suppose? but also fine to keep as is to match the style of `Variable` above